### ISSUE TITLE
Multiclass training?

### ISSUE THREAD

**Issue by ibarrien** [princeton-nlp/SimCSE#68]
Hi! Suppose we have a file that has two columns: input[str], label[int]. Here "input" is a sentence and "label" is a number between 0 to 5. How may I use your repo to build a classifier in this case? E.g. using supervised constrastive learning for pre-training, followed by classification training?

**Comment by gaotianyu1350** [comment]
Hi,

I assume you are talking about a regression task with only one sentence as input. In this case, you can directly use our pre-trained checkpoint (see readme) and fine-tune the model on the supervision data. About how to fine-tune the model on a supervised task, you can refer to [Huggingface's repo](https://github.com/huggingface/transformers).

**Comment by ibarrien** [comment]
Hi, thanks for your response.
I’m not asking about regression, but rather a single label multi class
problem where there the label space is 6 dimensional (0,1,2,3,4,5).

**Comment by gaotianyu1350** [comment]
I see. Then you can try to pair the sentences with the same label as positive pairs and do supervised contrastive learning.
